src/proposals.rs

```diff
@@ -0,0 +1,269 @@
+use std::time::Duration;
+
+use abscissa_core::{
+    tracing::{debug, error, info, log::warn},
+    Application,
+};
+use gravity_bridge::gravity_proto::{
+    cosmos_sdk_proto::cosmos::base::tendermint::v1beta1::GetLatestBlockRequest,
+    gravity::DelegateKeysByOrchestratorRequest,
+};
+use somm_proto::cosmos_sdk_proto::cosmos::gov::v1beta1::QueryProposalRequest;
+use tokio::{sync::mpsc::Sender, task::JoinHandle};
+use tonic::{transport::Channel, Code};
+
+use crate::{
+    config::get_delegate_address,
+    cork::{client::CorkQueryClient, id_hash, proposals::handle_scheduled_cork_proposal},
+    error::{Error, ErrorKind},
+    prelude::APP,
+    pubsub::cache::refresh_publisher_trust_state_cache,
+};
+
+type GovQueryClient = gravity_bridge::gravity_proto::cosmos_sdk_proto::cosmos::gov::v1beta1::query_client::QueryClient<Channel>;
+type GravityQueryClient =
+    gravity_bridge::gravity_proto::gravity::query_client::QueryClient<Channel>;
+type TendermintQueryClient = gravity_bridge::gravity_proto::cosmos_sdk_proto::cosmos::base::tendermint::v1beta1::service_client::ServiceClient<Channel>;
+
+// Proposal types
+const ADD_PUBLISHER_PROPOSAL: &str = "/pubsub.v1.AddPublisherProposal";
+const REMOVE_PUBLISHER_PROPOSAL: &str = "/pubsub.v1.RemovePublisherProposal";
+// const ADD_DEFAULT_SUBSCRIPTION_PROPOSAL: &str = "/pubsub.v1.AddDefaultSubscriptionProposal";
```

Comment:
I think we may want to add these back (and below) so that we do refresh the cache when a new default subscription is added.